hail_scripts/reference_data/combine.py

```diff
@@ -114,21 +125,43 @@ def join_hts(datasets, version, reference_genome='37'):
     )
 
     # NB: coverage datasets are keyed by locus rather than locus
-    # and alleles, so we cannot join.  Instead we annotate w/ locus as
+    # and alleles, so we cannot join.  Instead we annotate w/ locus
     # as the key.
     coverage_hts = [
         (dataset, get_ht(dataset, reference_genome))
         for dataset in datasets
         if 'coverage' in dataset
     ]
     for dataset, coverage_ht in coverage_hts:
-        joined_ht.annotate(dataset=coverage_ht[coverage_ht.locus][dataset])
+        joined_ht = joined_ht.annotate(
+            **{dataset: coverage_ht[joined_ht.locus][dataset]},
+        )
 
-    joined_ht = update_joined_ht_globals(
+    return update_joined_ht_globals(
         joined_ht,
         datasets,
         version,
         reference_genome,
     )
-    joined_ht.describe()
-    return joined_ht
+
+
+def update_existing_joined_hts(
+    destination_path: str,
+    dataset: str,
+    datasets: List[str],
+    version: str,
+    genome_version: str,
+):
+    joined_ht = hl.read_table(destination_path)
+    dataset_ht = get_ht(dataset, genome_version)
+    if 'coverage' not in dataset:
+        joined_ht = joined_ht.drop(dataset)
+        joined_ht = joined_ht.join(dataset_ht, 'outer')
+        joined_ht = joined_ht.filter(
+            hl.any([~hl.is_missing(joined_ht[dataset]) for dataset in datasets]),
+        )
+    else:
+        joined_ht = joined_ht.annotate(
+            **{dataset: dataset_ht[joined_ht.locus][dataset]},
+        )
```

Comment:
Possibly add a checkpoint here?

Reply:
I added the checkpoint to the last line here...

I'm actually seeing some strange behavior in the tests here...

Doing

```
print(joined_ht.collect())
joined_ht = joined_ht.checkpoint(checkpoint_path, overwrite=True, stage_locally=True)
print(joined_ht.collect())
```

is giving me different results... with the checkpoint seemingly reverting the applied join.  What am I doing wrong here?

Reply:
So the first print statement shows the newly added ht annotations and the second one is not?

Reply:
yep!  that's exactly it.  I'm gonna come back to this and see if I can re-produce in a shell.